### DISCUSSION TITLE
(use case): Use shields.io syntax for card parameters

### DISCUSSION THREAD

**Discussion by ghost** [anuraghazra/github-readme-stats#2327] [Q&A] [answered]
Hi everyone.

I wonder if you've all thought of joining github-readme-stats to shields.

#### url *before*
```
https://github-readme-stats.vercel.app/api?username=anuraghazra&count_private=true
```

#### url *after*
```
https://img.github-readme-stats.vercel.app/api/<USERNAME>-<COUNT_PRIVATE>
```

#### url *before*
```
https://github-readme-stats.vercel.app/api?username=anuraghazra&show_icons=true&theme=radical
```

#### url *after*
```
https://img.github-readme-stats.vercel.app/api/<USERNAME>-<SHOW_ICONS>-<THEME>
```

#### What is this feature?
My general idea is to follow the shields pattern in github-readme-stats. Before the github-readme-stats domain add the **img**.

#### Why is this feature interesting?
- Usually for me and a group of programmers it's common to add metadata in README.md files on GitHub. So... when adding this metadata... usually I add shields that are responsible for creating icons.  So... I would be happy if github-readme-stats followed the same url pattern, shields api, it would make my life easier - partly because I would do the same thing as shields in github-readme-stats.
- Another good reason is that shield users would get used to using github-readme-stats in the same way - This could improve the using experience of the two projects: shields and github-readme-stats

*Just a thought.*

**Comment by rickstaa** [comment, marked as answer]
@codehangen Thanks for your suggestion. I checked the [shields.io](https://shields.io/) syntax, and it looks like it also uses the [regular query parameter syntax](https://www.abstractapi.com/api-glossary/query-parameters):

```
https://img.shields.io/static/v1?label=<LABEL>&message=<MESSAGE>&color=<COLOR>
```

The only different thing is the `img` prefix. Furthermore, to my knowledge, the syntax you suggest is not recommended for Vercel projects (see https://vercel.com/blog/vercel-node-helpers). I would therefore stay with the current syntax since it is more widely used.

**Reply by ghost** [reply]
Hi rickstaa.

> @codehangen Thanks for your suggestion. I checked the [shields.io](https://shields.io/) syntax, and it looks like it also uses the [regular query parameter syntax](https://www.abstractapi.com/api-glossary/query-parameters):
> ``https://img.shields.io/static/v1?label=<LABEL>&message=<MESSAGE>&color=<COLOR>``
> The only different thing is the img prefix. Furthermore, to my knowledge, the syntax you suggest is not recommended for Vercel projects (see https://vercel.com/blog/vercel-node-helpers). I would therefore stay with the current syntax since it is more widely used.

thank you for feedback.
